tests/integration/dbapi/async/V2/test_server_async.py

```diff
@@ -0,0 +1,76 @@
+import time
+from random import randint
+from typing import Callable
+
+from firebolt.db import Connection
+
+LONG_SELECT = "SELECT checksum(*) FROM GENERATE_SERIES(1, 2500000000)"  # approx 3 sec
+
+
+async def test_insert_async(connection: Connection) -> None:
+    cursor = connection.cursor()
+    rnd_suffix = str(randint(0, 1000))
+    table_name = f"test_insert_async_{rnd_suffix}"
+    try:
+        await cursor.execute(f"CREATE TABLE {table_name} (id INT, name TEXT)")
+        await cursor.execute_async(
+            f"INSERT INTO {table_name} (id, name) VALUES (1, 'test')"
+        )
+        token = cursor.async_query_token
+        assert token is not None, "Asyc token was not returned"
+        # sleep for 2 sec to make sure the async query is completed
+        time.sleep(2)
+        assert await connection.is_async_query_running(token) == False
+        assert await connection.is_async_query_successful(token) == True
+        # Verify the result
+        cursor = connection.cursor()
+        await cursor.execute(f"SELECT * FROM {table_name}")
+        result = await cursor.fetchall()
+        assert result == [[1, "test"]]
+    finally:
+        await cursor.execute(f"DROP TABLE {table_name}")
+
+
+async def test_insert_async_running(connection: Connection) -> None:
+    cursor = connection.cursor()
+    rnd_suffix = str(randint(0, 1000))
+    table_name = f"test_insert_async_{rnd_suffix}"
+    try:
+        await cursor.execute(f"CREATE TABLE {table_name} (id LONG)")
+        await cursor.execute_async(f"INSERT INTO {table_name} {LONG_SELECT}")
+        token = cursor.async_query_token
+        assert token is not None, "Asyc token was not returned"
+        assert await connection.is_async_query_running(token) == True
+        assert await connection.is_async_query_successful(token) is None
+    finally:
+        await cursor.execute(f"DROP TABLE {table_name}")
```

Comment:
Just curious what will happen if we have a running insert in async mode and will try to drop the table? Probably it should just wait for a table lock to get released or something

Reply:
Tested it via the API. `INSERT INTO dummy SELECT checksum(*) FROM GENERATE_SERIES(1, 250000000000) ` would keep on running, despite table `dummy` no longer being there. Though if there's continuous insert it might behave differently, but I have no easy way of reproducing that.